extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+    mongoclient = pymongo.MongoClient("mongodb://localhost:27017/")
+    db = mongoclient.vus.variant
+    
+    # if DNnr already in mongoDB, check if updated an replace if neccesary
+    if db.count_documents({"dn_no": patient_dn_no}) > 0:
+        logger.info('dn_no: %s already present' % patient_dn_no)
+        # get and parse "lastUpdatedOn" from MongoDB/Alissa for specific patient_dn_no
+        lastUpdatedOn_mongoDB = db.find_one({"dn_no" : patient_dn_no}, {"lastUpdatedOn":1, "_id":0})
+        for key, value in lastUpdatedOn_mongoDB.items():
+            lastUpdatedOn_mongoDB = value
+        lastUpdatedOn_Alissa = ONB01_analysis['lastUpdatedOn']
+        if lastUpdatedOn_Alissa == lastUpdatedOn_mongoDB:
+            logger.info('%s already in database, lastUpdatedOn Alissa is the same' % patient_dn_no) 
+            return
+        elif lastUpdatedOn_Alissa > lastUpdatedOn_mongoDB:
+            db.delete_many({"dn_no": patient_dn_no})
+            logger.info('removed: %s from database, start replacing with newer version' % patient_dn_no)
+        elif lastUpdatedOn_Alissa < lastUpdatedOn_mongoDB:
+            logger.info('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+            sys.exit('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+        else:
+            logger.info('unknown error lastUpdatedOn for patient %s, this should not happen' % patient_dn_no)
+            sys.exit('unknown error lastUpdatedOn for patient %s, this should not happen' % patient_dn_no)
+
+    # get all relevant info from one patient into one dictionary.
+    patient = {}
+    patient["dn_no"] = patient_dn_no
+    logger.info('start uploading to mongodb: %s' % patient_dn_no)
+    patient["patient_accession_no"] = accession_number
+    patient.update(ONB01_analysis)
+    patient["analysis"] = patient.pop("reference")
+    patient["platform_dataset"] = analyis_sources["platformDataSet"]["info"]
+
+    # changes the format of the annotation sources from Alissa, to fit in MongoDB and resemble the old O-schijf format
+    externalSources_dict = {}
+    variants = VUS_response
+    for key in analyis_sources['annotationSources']:
+        source_name = ""
+        source_value = ""
+        for x,y in key.items():
+            if x == 'name':
+                source_name = y
+            elif x == 'description':
+                source_value = y
+            else:
+                continue
+        if source_name != "" and source_value != "":
+            externalSources_dict[source_name] = source_value
+        else:
+            print("error, externalsources has changed format in Alissa")
+    patient["annotation_sources"] = externalSources_dict  # add previous section to total patient info
+    
+    # add VUS info to patientdata
+    for item in variants:
+        variant = item
+        variant.update(patient)
+        db.insert_one(variant)
```

Comment:
```python
for variant in variants:
    variant.update(patient)
    db.insert_one(variant)
```

Reply:
I think I tried that, but though it wouldn't work in the loop. I will check it again, because it is nicer if it works.

Reply:
worked :). I guess I tried too many things at once at some point 👍